_internals/software-architecture.md

```diff
@@ -5,10 +5,10 @@ nav_order: 1
 use_mermaid: true
 ---
 
-Babelfish implements Microsoft SQL Server behavior on top of PostgreSQL. To achieve this goal, a
-couple of areas had to be modified inside the PostgreSQL core. PostgreSQL does
-not offer this capability out-of-the-box, therefore add-ons had to be provided
-to make this happen.
+Babelfish implements Microsoft SQL Server behavior on top of PostgreSQL.
+To achieve this goal, the server had to be modified.  This was mostly done
+with [extensions](https://www.postgresql.org/docs/current/extend-extensions.html),
+keeping modifications to PostgreSQL core at a minimum.
```

Comment:
Line 11 is missing an article:

"keeping modifications to the PostgreSQL core at a minimum."

Reply:
Hm. To me, "PostgreSQL core" is a kind of name. Perhaps I should capitalize "core"?
You would not say "to the Amazon Aurora", would you?

Reply:
I would say 'to the Amazon Aurora core' - I see the core kind of like a body part...

Reply:
I added the article in 45d03e4d81154fe8c19ff42ada2dc470aa1a4780.
It is fun to discuss language with you &ndash; I like the "body part".